tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/symbols/TypeDefPackageIdTest.java

```diff
@@ -0,0 +1,191 @@
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.semantic.api.test.symbols;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.impl.symbols.AbstractTypeSymbol;
+import io.ballerina.compiler.api.impl.symbols.BallerinaTypeDefinitionSymbol;
+import io.ballerina.compiler.api.symbols.SymbolKind;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.Project;
+import org.ballerinalang.model.elements.PackageID;
+import org.ballerinalang.test.BCompileUtil;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.assertBasicsAndGetSymbol;
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDefaultModulesSemanticModel;
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDocument;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Test cases for user defined type packageIDs.
+ *
+ * @since 2201.9.0
+ */
+public class TypeDefPackageIdTest {
+
+    protected Project project;
+    private SemanticModel model;
+
+    @BeforeClass
+    public void setup() {
+        project = BCompileUtil.loadProject("test-src/symbols/symbol-package-id");
+        model = getDefaultModulesSemanticModel(project);
+    }
+
+    @Test(dataProvider = "TypeDefPackageIDPosProvider")
+    public void testTypeDefPackageId(int line, int offset, String name, String srcFile, String typeDescPackageID,
+                                     String internalPackageID) {
+        Optional<Document> srcDocument = getDocument(project, null, srcFile);
+        assertTrue(srcDocument.isPresent());
+
+        BallerinaTypeDefinitionSymbol symbol =
+                (BallerinaTypeDefinitionSymbol) assertBasicsAndGetSymbol(model, srcDocument.get(), line, offset, name,
+                        SymbolKind.TYPE_DEFINITION);
+        PackageID actualTypeDescPackageID = ((AbstractTypeSymbol) (symbol).typeDescriptor()).getBType().tsymbol.pkgID;
+        PackageID actualInternalPackageID = symbol.getInternalSymbol().pkgID;
+        assertEquals(actualTypeDescPackageID.toString(), typeDescPackageID);
+        assertEquals(actualInternalPackageID.toString(), internalPackageID);
+    }
+
+    @DataProvider(name = "TypeDefPackageIDPosProvider")
+    public Object[][] getTypeDefPackageIDPos() {
+        return new Object[][]{
+
```

Comment:
Can we add some tests referring to builtin types also?

Also will the behaviour now be different for

```ballerina
type T1 future;
```

vs 

```ballerina
type T1 future<any|error>;
```

?

In a way, that's OK I guess. The first is more like an alias.

As discussed with @prakanth97 also though, I think when we have

```
type T some-typedesc;
```

only `T` should have an owner, not `some-typedesc` generally. But we don't have the ideal separation atm to do this.

Reply:
Added test case for `future` builtin type in d63b2090d7d94931913c1329c4dfc5784e5d5164.

The behavior is different for the given scenario,

```ballerina
type T1 future;
type T2 future<any|error>;
```

For T1, the `PackageID` is still `ballerina/lang.annotations:0.0.0`. But for T2, it is `sample_org/symbol_package_id:0.1.0`. Please let me know if this is not the expected behavior.